### PR TITLE
Staging Sites: add staging site to hosting configuration submenu

### PR DESCRIPTION
**Pull request by katinthehatsite**
- Close https://github.com/Automattic/dotcom-forge/issues/1909

## Proposed Changes

* This PR adds **Staging site** item to the Hosting Configuration submenu on SMP

## Testing Instructions

*  Create a staging site from **Settings > Hosting Configuration** of an Atomic site
*  Go to `/sites`
* Click on the ellipsis menu by an Atomic site:

Screenshot 2023-03-09 at 1 40 54 PM

* Click on **Hosting Configuration** > **Staging site**:

Screenshot 2023-03-09 at 1 42 03 PM

* Check if it brings you to the **staging card** under `hosting-config`

**Testing instructions for a staging site**

*  Create a staging site from **Settings > Hosting Configuration** of an Atomic site
*  Go to `/sites`
* Click on the ellipsis menu by a staging site site
* Click **Hosting Configuration**. The **Staging site** submenu item should not be available

Screenshot 2023-03-09 at 1 45 43 PM

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/sites-dashboard/components/sites-ellipsis-menu.tsx [hunk 1]
```diff
@@ -243,8 +243,43 @@ const SiteDropdownMenu = styled( DropdownMenu )( {
 function useSubmenuItems( site: SiteExcerptData ) {
 	const { __ } = useI18n();
 	const siteSlug = site.slug;
+	const stagingSite = site.is_wpcom_staging_site;
```
Comment (sejas) [inline client/sites-dashboard/components/sites-ellipsis-menu.tsx]: Could we call it `isStagingSite` ?

If the variable is a boolean, I like to define variable names starting with `isSomething` or `hasSomthing` where the _answer_ is yes or no.

```suggestion
	const isStagingSite = site.is_wpcom_staging_site;
```

File: client/sites-dashboard/components/sites-ellipsis-menu.tsx [hunk 2]
```diff
@@ -243,8 +243,43 @@ const SiteDropdownMenu = styled( DropdownMenu )( {
...
+				{
+					label: __( 'Web server logs' ),
+					href: `/hosting-config/${ siteSlug }#web-server-logs`,
+					sectionName: 'logs',
+				},
+			];
+		}
 		return [
 			{
 				label: __( 'SFTP/SSH credentials' ),
```
Comment (sejas) [inline client/sites-dashboard/components/sites-ellipsis-menu.tsx:287]: This works well  for the case where we have two options. Is staging or not.

Could we refactor it to use
https://github.com/Automattic/wp-calypso/pull/73356/files#diff-173bec7d847197dc6a90c47661016102ba95728dab618e17b774c11c7ed1994cR258-R310

File: client/sites-dashboard/components/sites-ellipsis-menu.tsx [hunk 3]
```diff
@@ -243,8 +244,44 @@ const SiteDropdownMenu = styled( DropdownMenu )( {
...
```
Comment (sejas) [inline client/sites-dashboard/components/sites-ellipsis-menu.tsx]: This works well. The only downside is that we are defining almost the same menu twice. I want to suggest an optimization. 

When I had to deal with multiple different conditions, I created this pattern with conditions and a filter.
Could you try to refactor your code and re-use it?

https://github.com/Automattic/wp-calypso/pull/73356/files#diff-173bec7d847197dc6a90c47661016102ba95728dab618e17b774c11c7ed1994cR258-R310

File: client/sites-dashboard/components/sites-ellipsis-menu.tsx [hunk 4]
```diff
@@ -243,8 +244,44 @@ const SiteDropdownMenu = styled( DropdownMenu )( {
 function useSubmenuItems( site: SiteExcerptData ) {
 	const { __ } = useI18n();
 	const siteSlug = site.slug;
+	const stagingSite = site.is_wpcom_staging_site;
+	const isStagingSiteEnabled = isEnabled( 'yolo/staging-sites-i1' );
```
Comment (sejas) [inline client/sites-dashboard/components/sites-ellipsis-menu.tsx:248]: Great usage of the feature flag 👏

**Review by sejas — CHANGES_REQUESTED** [review]
Tested and working well. 🥳 
It also scrolls the page to the correct section 👌 .

I have included two suggestions to enhance it further.

File: client/sites-dashboard/components/sites-ellipsis-menu.tsx (hunk 3 above)
Comment (katinthehatsite) [reply client/sites-dashboard/components/sites-ellipsis-menu.tsx]: It took me a bit of tinkering around but I think it looks better now - let me know what you think!

**Review by sejas — APPROVED** [review]
@katinthehatsite , Awesome!
Thanks for the changes. It works as expected, let's deploy it! 🥳

**Pull request merged**